kernel/h1_syscalls/src/fuse.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 lowRISC contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use byteorder::ByteOrder;
+use core::cell::Cell;
+use core::mem;
+use h1::hil::fuse::Fuse;
+use kernel::{AppId, Callback, Driver, Grant, ReturnCode, Shared, AppSlice};
+
+pub const DRIVER_NUM: usize = 0x40050;
+
+#[derive(Default)]
+pub struct AppData {
+    dev_id_buffer: Option<AppSlice<Shared, u8>>,
+}
+
+pub struct FuseSyscall<'a> {
+    fuse: &'a dyn Fuse,
+    apps: Grant<AppData>,
+    current_user: Cell<Option<AppId>>,
+}
+
+impl<'a> FuseSyscall<'a> {
+    pub fn new(fuse: &'a dyn Fuse,
+               container: Grant<AppData>) -> FuseSyscall<'a> {
+        FuseSyscall {
+            fuse: fuse,
+            apps: container,
+            current_user: Cell::new(None),
+        }
+    }
+
+    fn get_dev_id(&self, caller_id: AppId) -> ReturnCode {
+        self.apps.enter(caller_id, |app_data, _| {
+            if let Some(ref mut dev_id_buffer) = app_data.dev_id_buffer {
+                let dev_id = self.fuse.get_dev_id();
+                let val_size = mem::size_of::<u64>();
+                if dev_id_buffer.len() < val_size {
+                    return ReturnCode::ENOMEM;
+                }
+
+                byteorder::BE::write_u64(&mut dev_id_buffer.as_mut(), dev_id);
```

Comment:
Adding the `byteorder` crate just for this one line seems to be overkill. It's probably better to leave out the dependency and use the standard library to do this:

```rust
for (i, &byte) in dev_id.to_be_bytes().iter().enumerate() {
    match dev_id_buffer.as_mut().get_mut(i) {
        None => return ReturnCode::ENOMEM,
        Some(value) => *value = byte,
    }
}
```

Reply:
I didn't actually know that there's support in the standard library for that, incl. `from_be_bytes`.  Thank you! Changed.